src/Microsoft.Data.SqlClient/tests/Common/LocalAppContextSwitchesHelper.cs

```diff
@@ -0,0 +1,355 @@
+            value = GetValue(field);
+        }
+
+        // Capture the original value of each switch.
+        InitField(
+            "s_legacyRowVersionNullBehavior",
+            out _legacyRowVersionNullBehaviorField,
+            out _legacyRowVersionNullBehaviorOriginal);
+
+        InitField(
+            "s_suppressInsecureTLSWarning",
+            out _suppressInsecureTLSWarningField,
+            out _suppressInsecureTLSWarningOriginal);
+
+        InitField(
+            "s_makeReadAsyncBlocking",
+            out _makeReadAsyncBlockingField,
+            out _makeReadAsyncBlockingOriginal);
+
+        InitField(
+            "s_useMinimumLoginTimeout",
+            out _useMinimumLoginTimeoutField,
+            out _useMinimumLoginTimeoutOriginal);
+
+        InitField(
+            "s_legacyVarTimeZeroScaleBehaviour",
+            out _legacyVarTimeZeroScaleBehaviourField,
+            out _legacyVarTimeZeroScaleBehaviourOriginal);
+
+        InitField(
+            "s_useCompatProcessSni",
+            out _useCompatProcessSniField,
+            out _useCompatProcessSniOriginal);
+
+        InitField(
+            "s_useCompatAsyncBehaviour",
+            out _useCompatAsyncBehaviourField,
+            out _useCompatAsyncBehaviourOriginal);
+
+#if NETFRAMEWORK
+        InitField(
+            "s_disableTNIRByDefault",
+            out _disableTNIRByDefaultField,
+            out _disableTNIRByDefaultOriginal);
+#endif
+    }
+
+    // Disposal restores all original switch values as a best effort.
+    public void Dispose()
+    {
+        List<string> failedFields = new();
+
+        void RestoreField(FieldInfo field, Tristate value)
+        {
+            try
+            {
+                SetValue(field, value);
+            }
+            catch (Exception)
+            {
+                failedFields.Add(field.Name);
+            }
+        }
+
+        RestoreField(
+            _legacyRowVersionNullBehaviorField,
+            _legacyRowVersionNullBehaviorOriginal);
+        RestoreField(
+            _suppressInsecureTLSWarningField,
+            _suppressInsecureTLSWarningOriginal);
+        RestoreField(
+            _makeReadAsyncBlockingField,
+            _makeReadAsyncBlockingOriginal);
+        RestoreField(
+            _useMinimumLoginTimeoutField,
+            _useMinimumLoginTimeoutOriginal);
+        RestoreField(
+            _legacyVarTimeZeroScaleBehaviourField,
+            _legacyVarTimeZeroScaleBehaviourOriginal);
+        RestoreField(
+            _useCompatProcessSniField,
+            _useCompatProcessSniOriginal);
+        RestoreField(
+            _useCompatAsyncBehaviourField,
+            _useCompatAsyncBehaviourOriginal);
+#if NETFRAMEWORK
+        RestoreField(
+            _disableTNIRByDefaultField,
+            _disableTNIRByDefaultOriginal);
+#endif
+        if (failedFields.Count > 0)
+        {
+            Assert.Fail(
+                $"Failed to restore the following fields: " +
```

Comment:
🤔 You're using an interpolated string here, but not interpolating anything...
This could all fit on one line as an interpolated string :)